stdlib/Statistics/src/moments.jl

```diff
@@ -0,0 +1,297 @@
+    var!(similar(A, Float64, Base.reduced_indices(axes(A), dim)), A, w, dim;
+         mean=mean, corrected=corrected)
+end
+
+##### Fused statistics
+"""
+    mean_and_var(x; [weights::AbstractWeights,] [dims,] corrected=false) -> (mean, var)
+
+Return the mean and variance of a real-valued array `x`, optionally over dimensions
+`dims`, as a tuple. Observations in `x` can be weighted using `weights`.
+Finally, bias correction is be applied to the variance calculation if `corrected=true`.
+See [`var`](@ref) documentation for more details.
+"""
+function mean_and_var(A::AbstractArray;
+                      weights::Union{AbstractWeights, Nothing}=nothing,
+                      dims=:,
+                      corrected::Bool=true)
+    m = mean(A, weights=weights, dims=dims)
+    v = var(A, mean=m, weights=weights, dims=dims, corrected=corrected)
+    m, v
+end
+
+"""
+    mean_and_std(x; [weights::AbstractWeights], [dims,] corrected=true) -> (mean, std)
+
+Return the mean and standard deviation of a real-valued array `x`,
+optionally over dimensions `dims`, as a tuple. Observations in `x`
+can be weighted using `weights`. Finally, bias correction is be applied to the
+standard deviation calculation if `corrected=true`.
+See [`std`](@ref) documentation for more details.
+"""
+function mean_and_std(A::AbstractArray;
+                      weights::Union{AbstractWeights, Nothing}=nothing,
+                      dims=:,
+                      corrected::Bool=true)
+    m = mean(A, weights=weights, dims=dims)
+    s = std(A, mean=m, weights=weights, dims=dims, corrected=corrected)
+    m, s
+end
+
+
+
+##### General central moment
+function _moment2(A::AbstractArray, m::Real; corrected=false)
+    s = 0.0
+    for i in eachindex(A)
+        @inbounds z = A[i] - m
+        s += z * z
+    end
+    varcorrection(length(A), corrected) * s
+end
+
+function _moment2(A::AbstractArray, w::AbstractArray, m::Real; corrected=false)
+    s = 0.0
+    for i in eachindex(A, w)
+        @inbounds z = A[i] - m
+        @inbounds s += (z * z) * w[i]
+    end
+
+    varcorrection(w, corrected) * s
+end
+
+function _moment3(A::AbstractArray, m::Real)
+    s = 0.0
+    for i in eachindex(A)
+        @inbounds z = A[i] - m
+        s += z * z * z
+    end
+    s / length(A)
+end
+
+function _moment3(A::AbstractArray, w::AbstractArray, m::Real)
+    s = 0.0
+    for i in eachindex(A, w)
+        @inbounds z = A[i] - m
+        @inbounds s += (z * z * z) * w[i]
+    end
+    s / sum(w)
+end
+
+function _moment4(A::AbstractArray, m::Real)
+    s = 0.0
+    for i in eachindex(A)
+        @inbounds z = A[i] - m
+        s += abs2(z * z)
+    end
+    s / length(A)
+end
+
+function _moment4(A::AbstractArray, w::AbstractArray, m::Real)
+    s = 0.0
+    for i in eachindex(A, w)
+        @inbounds z = A[i] - m
+        @inbounds s += abs2(z * z) * w[i]
+    end
+    s / sum(w)
+end
+
+function _momentk(A::AbstractArray, k::Int, m::Real)
```

Comment:
These are not nice,
can't we do something better than this?
even if it is `@generated _moments(::Val{K}, A::AbstractArray, w::AbstractArray, m::Real) where K`

Reply:
I guess so, but let's decide whether we want to keep this function first (see OP). Is it really common to compute moments beyond order 4 (kurtosis)?

Reply:
Sorry I was unclear.
I was suggesting replacing monents 1-4 with something that wasn't hand coded.
(Though if you didn't want to go beyond 4, can use `@eval` I guess)